modules/stochastic_tools/src/reporters/AdaptiveImportanceStats.C

```diff
@@ -0,0 +1,101 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "AdaptiveImportanceStats.h"
+#include "Sampler.h"
+#include "Distribution.h"
+#include "Normal.h"
+#include "libmesh/utility.h"
+
+registerMooseObject("StochasticToolsApp", AdaptiveImportanceStats);
```

Comment:
Just a thought. What if we added this in the `AdaptiveMonteCarloDecision` instead of creating new object? `AdaptiveMonteCarloDecision` is already a reporter so we can create these reporter values based on whether the parameters are specified by the user. You can let me know if I'm crazy.

Reply:
For now, as per our discussion, I didn't make any changes to have a generalized statistics class. We should have a discussion with the team on how this will work for both regular and adaptive Monte Carlo sampling since the mechanism to compute the statistics are different across different methods. Please let me know what you think.